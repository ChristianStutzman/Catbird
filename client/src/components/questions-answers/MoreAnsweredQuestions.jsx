import React from 'react';
import { useSelector } from 'react-redux';

const MoreAnsweredQuestions = () => {

  return (
    <div>

    </div>
  )
}

export default MoreAnsweredQuestions