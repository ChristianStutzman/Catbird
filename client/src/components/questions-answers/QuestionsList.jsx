import React from 'react';
import IndividualQuestion from './individual-question/Index.jsx';
import { Container, Button, Row, Col } from 'react-bootstrap';
import AddQuestion from './add-question-model/Index.jsx';
import ListGroup from 'react-bootstrap/ListGroup'



const QuestionsList = ({ qnaList, getMoreQuestions, noMoreQuestion }) => {

     let qList = qnaList.map(question => {
     return (
      <ListGroup.Item key={'QuestionsList' + question.question_id} >

        <IndividualQuestion question={question} />
      </ListGroup.Item>

      )
  })

  return (
    <Container>
      <ListGroup variant="flush">
        {qList}
      </ListGroup>

      <br />
      <Container className="twoMainButton">
       <Row className="flex-nowrap text-center">

         {!noMoreQuestion &&
            <Col className="flex-md-fill">
            <Button variant="outline-primary" size="sm" onClick={getMoreQuestions} >MORE ANSWERED QUESTION</Button>
            </Col>
         }

          <Col className="flex-md-fill">
            <AddQuestion />
          </Col>
       </Row>
       </Container>

    </Container>
  )


}

export default QuestionsList;