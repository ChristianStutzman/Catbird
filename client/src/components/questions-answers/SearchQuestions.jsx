import React, { useState } from 'react';
import SearchIcon from "@material-ui/icons/Search";
import { useSelector, useDispatch } from 'react-redux';
import IndividualQuestion from './individual-question/Index.jsx';
import { Container, ListGroup, Row, Col } from 'react-bootstrap';


const SearchQuestions = () => {

  const qnaList = useSelector(state => state.qnaList);
  const searchBarTyped = useSelector(state => state.searchBarTyped);
  const dispatch = useDispatch();
  const [filteredData, setFilteredData] = useState([]);
  const [ wordEntered, setWordEntered ] = useState('');

  const handleSearch = (e) => {

    e.preventDefault();
    const searchWord = e.target.value;
    setWordEntered(searchWord);

    if (e.target.value.length !== 0) {
      dispatch({
        type:'SEARCHBAR_START_TYPING',
        searchBarTyped: true
      })
    }

    if (e.target.value.length === 0) {
      dispatch({
        type:'SEARCHBAR_START_TYPING',
        searchBarTyped: false
      })
    }

  const newFilter = qnaList.filter((value) => {
    return value.question_body.toLowerCase().includes(searchWord.toLowerCase());
  });



  if (searchWord === "") {
    setFilteredData([]);
  } else {
    setFilteredData(newFilter);
  }


}





  let filteredqList = filteredData.map(question => {
    return (
     <ListGroup.Item key={question.question_id} >

       <IndividualQuestion question={question} />

     </ListGroup.Item>

     )
 })


  return (

    <Container>

      <Row>

        <form className="form">
          <input
            className="search-input"
            placeholder='HAVE A QUESTION? SEARCH FOR ANSWERS...'
            onChange={handleSearch}
            />
          <SearchIcon />
         </form>

      </Row>



      <Row>

      { searchBarTyped ?
            <Container className="searchedQuestions">
            <ListGroup variant="flush" className="searchedQuestions" >
              {filteredqList}
            </ListGroup>
            <br />
            </Container> : null
      }


      </Row>



    </Container>
  )
}


export default SearchQuestions;
