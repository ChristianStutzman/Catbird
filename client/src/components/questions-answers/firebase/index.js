import firebase from "firebase/app";
import "firebase/storage";




const firebaseConfig = {
  apiKey: "AIzaSyAqHkilEykybKd9VL7QPLlwtEkE_Zu0XwQ",
  databaseURL: "gs://sunlit-analyst-318318.appspot.com/",
  authDomain: "sunlit-analyst-318318.firebaseapp.com",
  projectId: "sunlit-analyst-318318",
  storageBucket: "sunlit-analyst-318318.appspot.com",
  messagingSenderId: "767876054954",
  appId: "1:767876054954:web:41dd6f1c026243e3b575d3",
  measurementId: "G-2FY9FY7FSX"
};

  firebase.initializeApp(firebaseConfig);

  const storage = firebase.storage();

  export { storage, firebase as default };