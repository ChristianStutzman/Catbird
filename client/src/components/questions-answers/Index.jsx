import React, { useEffect} from 'react';
import { useDispatch, useSelector} from 'react-redux';
import { bindActionCreators } from 'redux';
import { actionCreators } from '../../redux/index.js';
import QuestionsList from './QuestionsList.jsx';
// import IndividualQuestion from './individual-question/Index.jsx';
import SearchQuestions from './SearchQuestions.jsx';
import MoreAnsweredQuestions from './MoreAnsweredQuestions.jsx';
import AddQuestion from './add-question-model/Index.jsx';
import AddAnswer from './add-answer-model/Index.jsx';
import axios from 'axios';

const QnAComponent = (props) => {
const state = useSelector(state => state);

useEffect(() => {
  axios.get('h')
})

  return (
     <div>
    <QuestionsList />
    <SearchQuestions />
    <MoreAnsweredQuestions />
    <AddQuestionModel />
    <AddAnswerModel />
    </div>

  )

}




export default QnAComponent;