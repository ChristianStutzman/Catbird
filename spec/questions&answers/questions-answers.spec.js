
import React from 'react';
import { render, screen, fireEvent, cleanup } from './test-utils';
import { questions } from '../../sampleData/questions-answers.js';
import QnAComponent from '../../client/src/components/questions-answers/Index.jsx';
import QuestionsList from '../../client/src/components/questions-answers/QuestionsList.jsx';


describe('root component render without error', () => {

  beforeEach(() => {
      render(<QnAComponent />)
    });

    afterEach(() => {
      cleanup();
    })

    test('render QnaComponent with redux', () => {
      const div = document.createElement('div');
      const component = render(<QnAComponent />, div)
      expect(component.container).toMatchSnapshot()
    })

})


describe('QuestionList component can render without error', () => {

  it('render 4 question on init', () => {
    const { getAllByTestId } = render(<QnAComponent qnaList={questions}/>, { initialState: { qnaList: questions }})
    expect(getAllByTestId('qList')).toHaveLength(4);

  })

})





