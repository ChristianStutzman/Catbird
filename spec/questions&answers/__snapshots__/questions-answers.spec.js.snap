// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with redux 1`] = `
<div>
  <div
    class="container"
  >
    <br />
    <div
      class="row"
    >
      <div
        class="col"
      />
      <div
        class="qna-title text-center col-5"
      >
        QUESTION AND ANSWERS
      </div>
      <div
        class="col"
      />
    </div>
    <br />
    <div
      class="flex-nowrap text-center container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <div>
              <form
                class="form"
              >
                <input
                  class="search-input"
                  placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root searchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </form>
            </div>
            <div
              class="filteredQuestionList"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="twoMainButton container"
      >
        <div
          class="flex-nowrap text-center row"
        >
          <div
            class="flex-md-fill col"
          >
            <div>
              <div>
                <button
                  class="btn btn-outline-primary btn-sm"
                  data-testid="addQuestion-btn"
                  type="button"
                >
                  ADD A QUESTION + 
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`root component render without error render QnaComponent with redux 1`] = `
<div>
  <div
    class="container"
  >
    <br />
    <div
      class="row"
    >
      <div
        class="col"
      />
      <div
        class="qna-title text-center col-5"
      >
        QUESTION AND ANSWERS
      </div>
      <div
        class="col"
      />
    </div>
    <br />
    <div
      class="flex-nowrap text-center container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <div>
              <form
                class="form"
              >
                <input
                  class="search-input"
                  placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root searchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </form>
            </div>
            <div
              class="filteredQuestionList"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="twoMainButton container"
      >
        <div
          class="flex-nowrap text-center row"
        >
          <div
            class="flex-md-fill col"
          >
            <div>
              <div>
                <button
                  class="btn btn-outline-primary btn-sm"
                  data-testid="addQuestion-btn"
                  type="button"
                >
                  ADD A QUESTION + 
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`root component render without error render with redux 1`] = `
<div>
  <div
    class="container"
  >
    <br />
    <div
      class="row"
    >
      <div
        class="col"
      />
      <div
        class="qna-title text-center col-5"
      >
        QUESTION AND ANSWERS
      </div>
      <div
        class="col"
      />
    </div>
    <br />
    <div
      class="flex-nowrap text-center container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <div>
              <form
                class="form"
              >
                <input
                  class="search-input"
                  placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..."
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root searchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </form>
            </div>
            <div
              class="filteredQuestionList"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="twoMainButton container"
      >
        <div
          class="flex-nowrap text-center row"
        >
          <div
            class="flex-md-fill col"
          >
            <div>
              <div>
                <button
                  class="btn btn-outline-primary btn-sm"
                  data-testid="addQuestion-btn"
                  type="button"
                >
                  ADD A QUESTION + 
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
